index.js

```diff
@@ -7,8 +7,9 @@ function createIndex (ldb, opts) {
     return {
       map: function (msgs, next) {
         var ops = []
-        msgs.forEach(function (msg) {
-          var res = opts.map(msg)
+        let pending = msgs.length
+        msgs.forEach(async function (msg) {
+          var res = await opts.map(msg)
```

Comment:
Will this work even if `opts.map` isn't an async function?

Reply:
yup! i tried it locally before starting on the pr, it just resolves

here's MDN on that case:
> If the value of the expression following the await operator is not a Promise, it's converted to a [resolved Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve).
>
> [await - MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#Description)

now that i've read about it, i'm uncertain if there are any side effects